hazelcast/src/main/java/com/hazelcast/datastore/ExternalDataStoreService.java

```diff
@@ -26,6 +27,17 @@
  */
 @Beta
 public interface ExternalDataStoreService extends AutoCloseable {
+
+    /**
+     * Tests external data store configuration.
+     *
+     * @param config name of the data store factory
+     * @return {@code true} if test was successful
+     * @throws Exception if the test operation fails
+     * @since 5.3
+     */
+    boolean testConnection(ExternalDataStoreConfig config) throws Exception;
```

Comment:
Why not?
```suggestion
    boolean testConnection(String name) throws Exception;
```

Reply:
The idea is to support both use-cases:
- test connection BEFORE datastore is added to member config.
- test connection of already added datastore